### ISSUE TITLE
0xLeveler - SuperPool `togglePause` does not work as intended

### ISSUE THREAD

**Issue by sherlock-admin3** [sherlock-audit/2024-08-sentiment-v2-judging#251]
0xLeveler

High

# SuperPool `togglePause` does not work as intended

## Summary 
There is a `togglePause()` function in the pool that is supposed to pause the function state , according to the natspec. 
```solidity
/// @notice Toggle pause state of the SuperPool
 function togglePause() external onlyOwner {
        if (Pausable.paused()) Pausable._unpause();
        else Pausable._pause();
    }
```
except the toggle function alone is not enough to implement pausing to the pool.

## Vulnerability Detail
The SuperPool contract inherits Pausable, but does not completely implement it, This is a common but disastrous issue that is pointed out in the `pausable.sol` Natspec
"This module is used through inheritance. It will make available the
modifiers `whenNotPaused` and `whenPaused`, which can be applied to the functions of your contract. `Note that they will not be pausable by simply including this module, only once the modifiers are put in place`."

## Impact
SuperPool `deposits`, `withdraws` and functions that may need to be paused to prevent loss of funds in Superpools due to problems in pools or the superPool itself, during or post-exploits cannot be paused

## Code Snippet
https://github.com/sherlock-audit/2024-08-sentiment-v2/blob/main/protocol-v2/src/SuperPool.sol#L164-L167

## Tool used
Manual Review

## Recommendation
Add `whenNotPaused` and `whenPaused` modifiers to functions you want to pause and unpause. Inheritance is not enough.

Duplicate of #270
